src/CustomMarshalers/src/System/Runtime/InteropServices/CustomMarshalers/EnumeratorViewOfEnumVariant.cs

```diff
@@ -0,0 +1,69 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Runtime.InteropServices.ComTypes;
+using System.Text;
+
+namespace System.Runtime.InteropServices.CustomMarshalers
+{
+    internal class EnumeratorViewOfEnumVariant : ICustomAdapter, IEnumerator
+    {
+        private readonly IEnumVARIANT _enumVariantObject;
+        private bool _fetchedLastObject;
+        private object[] _nextArray = new object[1];
```

Comment:
Can this be static?

Reply:
I don't believe this can be static. I'm pretty sure that would cause tons of threading issues.